### PR TITLE
Move F3DOptions to Applications and improve f3d::options

### PR DESCRIPTION
**Pull request by mwestphal**
- Move F3DOptions to Applications/F3DOptionsParser
- Rewrite F3DOption to not use VTK at all and use filesystem instead
- Improve f3d::options for easier usages
- Use gcc-8 for filesystem support
- Add and update tests

### CODE REVIEW

File: .github/docker/Dockerfile [hunk 1]
```diff
@@ -17,10 +17,15 @@ RUN apt update && apt install -y \
     openssh-client \
     software-properties-common \
     wget \
-    xvfb 
+    xvfb \
+    gcc-8 \
```
Comment (Meakk) [inline .github/docker/Dockerfile:21]: sort the new packages

File: src/application/F3DOptionsParser.cxx [hunk 2]
```diff
@@ -179,101 +174,101 @@ class ConfigurationOptions
...
-F3DOptions ConfigurationOptions::GetOptionsFromArgs(std::vector<std::string>& inputs)
+void ConfigurationOptions::GetOptionsFromArgs(
+  F3DAppOptions& appOptions, f3d::options& options, std::vector<std::string>& inputs)
 {
-  F3DOptions options;
   try
   {
-    cxxopts::Options cxxOptions(f3d::AppName, f3d::AppTitle);
+    // TODO From where ?
+    cxxopts::Options cxxOptions("F3D", "F3D");
```
Comment (Meakk) [inline src/application/F3DOptionsParser.cxx:183]: Can you add back f3d::AppName and f3d::AppTitle in an application-side CMake generated file?

File: src/library/f3d_engine.cxx [hunk 3]
```diff
@@ -59,6 +59,12 @@ engine::~engine()
   delete this->Internals;
 }
 
+//----------------------------------------------------------------------------
+void engine::setOptions(options& opt)
```
Comment (Meakk) [inline src/library/f3d_engine.cxx:63]: Sorry I don't like a `setOptions` function making the input invalid. In this case, I would prefer to explicitly call `engine.getOptions() = std::move(myOptions);`.  
Alternatively, you can have these two functions:
```cpp
void engine::setOptions(const options& opt); // call copy
void engine::setOptions(options&& opt); // call move
```
Having both will allow the user to call `engine.setOptions(std::move(myOptions))` if he wants to move it.

**Review by Meakk — CHANGES_REQUESTED** [review]

File: src/library/f3d_config.h [hunk 4]
```diff
@@ -13,7 +13,5 @@ extern const std::string LibVersion;
 extern const std::string LibBuildSystem;
 extern const std::string LibBuildDate;
 extern const std::string LibCompiler;
-extern const std::string ReservedString; // TODO to remove once not necessary
```
Comment (Meakk) [inline src/library/f3d_config.h:16]: `AppName` and `AppTitle` can be removed

File: src/library/f3d_engine.h [hunk 5]
```diff
@@ -46,6 +46,16 @@ class F3D_EXPORT engine
...
 
+  // Engine provide a default options
+  // use this setter to use other options
+  // copy options into engine
+  void setOptions(const options& opt);
+
+  // Engine provide a default options
+  // use this setter to use other options
+  // move options into engine
+  void setOptions(options&& opt); // call move
```
Comment (Meakk) [inline src/library/f3d_engine.h:57]: remove `//call move`

**Review by Meakk — APPROVED** [review]
minor comments, lgtm

File: src/library/f3d_config.h (hunk 4 above)
Comment (mwestphal) [reply src/library/f3d_config.h:16]: not yet, no. I'm working on that on another branch.

**Pull request merged**